source/News.cpp

```diff
@@ -44,15 +44,28 @@ void News::Load(const DataNode &node, const ConditionsStore *playerConditions)
 
 		if(tag == "location")
 		{
-			if(remove)
+			if(add)
+			{
+				child.PrintTrace("Skipping 'add location'; adding to locationFilters is not supported:");
+				continue;
+			}
+			else if(remove)
```

Comment:
These `continue`s are redundant.

Also, the traces could be more generic. We don't need to specify the entire node in the print (e.g. "add location", "add to show") since the PrintTrace already puts the entire node in the errors.txt.

I think the messages I use for #11340 are sufficiently generic. I also think we shouldn't necessarily ignore `add location` if there was never any location to begin with. I imagine that's less likely to cause issues with existing plugins.
```suggestion
			if(add && !location.IsEmpty())
				child.PrintTrace("Error: Cannot \"add\" to an existing location filter:");
			else if(remove)
```